### DISCUSSION TITLE
Generate classes from existing database

### DISCUSSION THREAD

**Discussion by Rechdan** [linq2db/linq2db#3400] [Q&A] [answered]
Pretty straight forward question: is there a way of generating classes from an existing database?

**Comment by sdanyliv** [comment, marked as answer]
Actually it is on main page [Now let's create a POCO class](https://github.com/linq2db/linq2db#now-lets-create-a-poco-class) - first sentence.

**Reply by Rechdan** [reply]
Thanks for pointing that out!

I did follow these stepes, but I can't find the `LinqToDB.Templates` folder, I'm using .NET 6.0.101 and I'm writing my code under Linux with VS Code, maybe that's the reason?

**Reply by sdanyliv** [reply]
Unfortunately under Linux it will not work. Only from Visual Studio or, I think, from Rider. We are working on tooling which is not based on T4 templates, but this functionality will arrive only with version 4.0.

So, generate model using Visual Studio, then you can use it in Linux environment.

**Reply by Rechdan** [reply]
So, if I install version `4.0.0-preview.9-fixed`, is there a way of making it work?

**Reply by sdanyliv** [reply]
No, feature is still in development and not committed into 4.0 branch: https://github.com/linq2db/linq2db/pull/3098

**Reply by Rechdan** [reply]
Got it, thanks!

**Reply by Rechdan** [reply]
@sdanyliv what you told me to do worked perfectly! I used VS2022 in my Host and copied the generated code to the actual project in my VS Code (that is connected to my Linux VM).
